dealroom_firestore_connector/__init__.py

```diff
@@ -156,6 +158,42 @@ def collection_exists(collection_ref: CollectionReference):
     return len(docs) > 0
 
 
+def get_history_doc_refs(
+    db: firestore.Client,
+    finalurl_or_dealroomid: str
+) -> Union[Tuple[DocumentReference], int]:
+    """Returns a DocumentReference based on the final_url field or dealroom_id
+    field.
+
+    Args:
+        db (firestore.Client): the client that will perform the operations.
+        finalurl_or_dealroomid (str): either a domain or a dealroom ID. Query documents that match this parameter.
+
+    Returns:
+        Tuple[DocumentReference]: a sequence of document references matching the input parameter.
+
+    Examples:
+        >>> db = new_connection(project=FIRESTORE_PROJECT_ID)
+        >>> doc_refs = get_history_refs(db, "dealroom.co")
+    """
+
+    collection_path = "history"
+    collection_ref = db.collection(collection_path)
+
+    if finalurl_or_dealroomid.isnumeric():
+        query_params = ["dealroom_id", "==", int(finalurl_or_dealroomid)]
+    else:
+        query_params = ["final_url", "==", finalurl_or_dealroomid]
+
+    query = collection_ref.where(*query_params)
+    docs = stream(query)
+    if docs == -1:
+        logging.error("Couldn't stream query.")
+        return ERROR
+
+    return tuple(doc.reference for doc in docs)
```

Comment:
Just curious why you preferred tuple over a list?

Reply:
Because they're immutable. Not sure if this feature adds value here, though